Backend/app/rider-platform/rider-app/Main/src/Domain/Action/UI/Select.hs

```diff
@@ -93,3 +107,20 @@ selectList estimateId = do
   checkIfEstimateCancelled estimate.id estimate.status
   selectedQuotes <- runInReplica $ QQuote.findAllByEstimateId estimateId
   pure $ SelectListRes $ map DQuote.makeQuoteAPIEntity selectedQuotes
+
+selectResult :: (EsqDBReplicaFlow m r) => Id DEstimate.Estimate -> m QuotesResultResponse
+selectResult estimateId = do
+  estimate <- runInReplica $ QEstimate.findById estimateId >>= fromMaybeM (EstimateDoesNotExist estimateId.getId)
+  searchRequest <- runInReplica $ QSearchRequest.findById estimate.requestId >>= fromMaybeM (EstimateDoesNotExist estimateId.getId)
+  if estimate.autoAssignEnabled
+    then do
+      mbBooking <- runInReplica $ QBooking.findAssignedByRiderId searchRequest.riderId
+      case mbBooking of
+        Just booking ->
+          return $ QuotesResultResponse {bookingId = Just booking.id, selectedQuotes = Nothing}
+        Nothing -> do
+          result <- selectList estimateId
```

Comment:
Instead of calling this function, just use
```  
selectedQuotes <- runInReplica $ QQuote.findAllByEstimateId estimateId
 pure $ SelectListRes $ map DQuote.makeQuoteAPIEntity selectedQuotes
  ```